opensfm/src/sfm/CMakeLists.txt

```diff
@@ -11,7 +11,12 @@ target_link_libraries(sfm
   PRIVATE
     foundation
 )
-target_include_directories(sfm PUBLIC ${EIGEN_INCLUDE_DIRS} ${CMAKE_SOURCE_DIR})
+target_include_directories(sfm
+    PUBLIC
+        ${EIGEN_INCLUDE_DIRS}
+        ${CMAKE_SOURCE_DIR}
+    PRIVATE
+        ${OpenMP_CXX_INCLUDE_DIR})
```

Comment:
When should we use public or private? I see openmp added in the private section here and in the public one in the akaze file below

Reply:
That's a really good question. I have been playing around with that a bit and tried to reduce the includes to a minimum. To avoid having `PRIVATE` includes in most `CMakeLists.txt`,  `foundation` now adds the includes/libary as `PUBLIC` and `akaze` and `vlfeat` are still `PRIVATE`.